# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../bird/0001-tgetent.patch
# Copyright (C) 2013 - 2014 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

ncurses can be build as separate libraries, thus it is possible to have a
separated libtinfo which has the symbol for tgetent.
This patch is changing the order of library checks to find tgetent, thus
that tinfo libs get checked before ncurses, so birdc will be linked against
libtinfo and not libncurses which leads to linking problems.


--- ./configure.in.orig   2014-04-01 12:24:10.000000000 +0300
+++ ./configure.in        2014-09-26 21:57:26.085053034 +0300
@@ -267,10 +267,10 @@
 if test "$enable_client" = yes ; then
	CLIENT=birdc
	AC_CHECK_LIB(history, add_history, CLIENT_LIBS="-lhistory")
-	AC_CHECK_LIB(ncurses, tgetent, USE_TERMCAP_LIB=-lncurses,
-		AC_CHECK_LIB(curses, tgetent, USE_TERMCAP_LIB=-lcurses,
-			AC_CHECK_LIB(tinfow, tgetent, USE_TERMCAP_LIB=-ltinfow,
-				AC_CHECK_LIB(tinfo, tgetent, USE_TERMCAP_LIB=-ltinfo,
+	AC_CHECK_LIB(tinfow, tgetent, USE_TERMCAP_LIB=-ltinfow,
+		AC_CHECK_LIB(tinfo, tgetent, USE_TERMCAP_LIB=-ltinfo,
+			AC_CHECK_LIB(ncurses, tgetent, USE_TERMCAP_LIB=-lncurses,
+				AC_CHECK_LIB(curses, tgetent, USE_TERMCAP_LIB=-lcurses,
					AC_CHECK_LIB(termcap, tgetent, USE_TERMCAP_LIB=-ltermcap,
						AC_MSG_ERROR([[The client requires ncurses library. Either install the library or use --disable-client to compile without the client.]]))))))
	AC_CHECK_LIB(readline, rl_callback_read_char, CLIENT_LIBS="-lreadline $CLIENT_LIBS $USE_TERMCAP_LIB",
